pyproject.toml

```diff
@@ -71,8 +71,13 @@ optional = [
     "scipy",
     "shapely >=1.8",
     "vtk",
-    "xmipy",
 ]
+
+api = [
+    "xmipy @ git+https://git@github.com/Deltares/xmipy@develop",
+    "modflowapi @ git+https://git@github.com/MODFLOW-USGS/modflowapi@develop"
```

Comment:
is it possible to not pin these to git develop versions? These pip dependencies are tricky to use, for instance, someone may have already done `pip install -e .` for (e.g.) a local develop clone of xmipy, but this would get uninstalled/replaced with a non-editable copy of the same repo. Ideally it should be pinned to a minimum version released on PyPI, which would normally leave a local develop clone alone if it is up-to-date.

Reply:
@mwtoews, this feature is dependent on versions of xmipy and modflowapi that have not been released yet.  Once those versions have been released these dependencies can be updated as you suggested.

Reply:
@w-bonelli, good idea.  Changed the PR so that FloPy now lets modflowapi fulfill the xmipy dependency.

Reply:
A better workaround is to unpin the explicit development version here and do a separate `pip install modflowapi @ ...` command in the CI workflow before installing `flopy[api]`. In related developments, [xmipy v1.3.0](https://github.com/Deltares/xmipy/releases/tag/v1.3.0) was released yesterday.

Reply:
@mwtoews, I apologize in advance since I am still learning about the .toml file and these CI workflows.  I do not understand why the develop version of modflowapi should be removed from pyproject.toml.  Shouldn't the pyproject.toml file reflect what needs to be installed for FloPy to function properly?

Note that the next modflowapi release will likely happen soon, maybe even before this FloPy plugins PR gets approved.  When it does I can remove all modflowapi develop version dependencies from pyproject.toml and all the workflows, which will solve the problem.  This is just a short-term solution so that I can get this feature into review without having to wait for the next release of modflowapi.

Reply:
The reason why the develop version of modflowapi should be unpinned from pyproject.toml is that it will overwrite any other local version of modflowapi, which I find disruptive (since I probably already have a develop version installed). It is overwritten since the develop version has no version number to compare. My suggestion is analogous to xmipy to first [install xmipy/modflowapi in the CI script](https://github.com/MODFLOW-USGS/modflowapi/blob/a8e241df1def5899ccbf22368eddc76da0d7a60c/.github/workflows/ci.yml#L124-L126), e.g. before `pip install .[api]`. The unpinned version should look like [this pyproject.toml](https://github.com/MODFLOW-USGS/modflowapi/blob/a8e241df1def5899ccbf22368eddc76da0d7a60c/pyproject.toml#L37) with a helpful comment. A runtime check is a bit more difficult for me to suggest, as I don't know what features are being used in the develop version, but it might be not necessary if a release is immanent. Further clarification is always welcome!

Reply:
Now that modflowapi version 0.1.0 has been released I have removed the develop version pip dependencies from flopy.  This should resolve the issue.